framework/db/Connection.php

```diff
@@ -811,13 +811,16 @@ function ($matches) {
      */
     public function getDriverName()
     {
-        if ($this->_driverName === null) {
-            if (($pos = strpos($this->dsn, ':')) !== false) {
-                $this->_driverName = strtolower(substr($this->dsn, 0, $pos));
-            } else {
-                $this->_driverName = strtolower($this->getSlavePdo()->getAttribute(PDO::ATTR_DRIVER_NAME));
-            }
+        if ($this->_driverName !== null) {
+            return $this->_driverName;
         }
+
+        $foundDSN = strstr($this->dsn, ':', true);
+        if ($foundDSN !== false) {
+            $foundDSN = $this->getSlavePdo()->getAttribute(PDO::ATTR_DRIVER_NAME);
+        }
+
+        $this->_driverName = strtolower($foundDSN);
```

Comment:
I'm interested to delete this line. 

I have connection error comes from PDO: "the driver can not be found", if it is in uppercase. 

:confused: